src/v/pandaproxy/rest/proxy.cc

```diff
@@ -9,52 +9,125 @@
 
 #include "pandaproxy/rest/proxy.h"
 
+#include "config/property.h"
+#include "config/rest_authn_endpoint.h"
+#include "net/unresolved_address.h"
 #include "pandaproxy/api/api-doc/rest.json.h"
 #include "pandaproxy/logger.h"
+#include "pandaproxy/parsing/exceptions.h"
+#include "pandaproxy/parsing/from_chars.h"
 #include "pandaproxy/rest/configuration.h"
 #include "pandaproxy/rest/handlers.h"
+#include "pandaproxy/types.h"
+#include "redpanda/request_auth.h"
+#include "utils/gate_guard.h"
 
 #include <seastar/core/future-util.hh>
 #include <seastar/core/memory.hh>
 #include <seastar/http/api_docs.hh>
 
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/constants.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <algorithm>
+#include <string>
+
 namespace pandaproxy::rest {
 
+namespace {
+net::unresolved_address address_from_url(std::string_view url) {
```

Comment:
> pp/handlers: wireup auth and cache on /brokers

it looks like this commit is doing a lot more than wiring up auth. generally 'wire up' is stuff like passing parameters down through a deep call stack or something.

i think this commit should be broken up if there are multiple things going on, or, the commit message needs to explain what is happening.